Simperium/src/main/java/com/simperium/android/CredentialsActivity.java

```diff
@@ -315,15 +315,21 @@ private boolean isValidEmail(String text) {
         return Patterns.EMAIL_ADDRESS.matcher(text).matches();
     }
 
+    // Password is valid if:
+    // - Meets minimum length requirement based on login (PASSWORD_LENGTH_LOGIN) and signup (PASSWORD_LENGTH_MINIMUM)
+    // - Does not have new lines or tabs (PATTERN_NEWLINES_TABS)
+    // - Does not match email address
     private boolean isValidPassword(String password) {
-        return isValidPasswordLength() && !PATTERN_WHITESPACE.matcher(password).find();
+        return isValidPasswordLength(mIsLogin) && !PATTERN_NEWLINES_TABS.matcher(password).find() &&
+            (mInputEmail.getEditText() != null && mInputPassword.getEditText() != null &&
+            !mInputEmail.getEditText().getText().toString().contentEquals(mInputPassword.getEditText().getText().toString()));
```

Comment:
I think it would be better to use the `password` parameter that is passed into this function for the email check instead of re-retrieving it from the input field.

Similarly, every time we call this method we have [already retrieved the email value from the input field](https://github.com/Simperium/simperium-android/pull/219/files#diff-f578252d8746b4d32cf359ef222428b4R407), if we passed that email value into this method instead of re-retrieving it here inside the method that would insure that we're using the same "email" in both places and simplify the logic here.

Reply:
Good idea!  I added the `email` parameter to `isValidPassword` in ef59a99.